polkadot/xcm/pallet-xcm/src/lib.rs

```diff
@@ -2954,6 +2955,59 @@ impl<T: Config> Pallet<T> {
 		})
 	}
 
+	/// Computes the weight cost using the provided `WeightTrader`.
```

Comment:
Should we say at the start of the doc comment that this function is supposed to ONLY be called by the `XcmPaymentApi`?